frontend/src/components/resourceMap/graph/graphModel.tsx

```diff
@@ -116,3 +122,82 @@ export interface Relation {
   toSource?: string;
   predicate: (from: GraphNode, to: GraphNode) => boolean;
 }
+
+/**
+ * Default node weight assignments for different Kubernetes resource types.
+ * Higher weight = higher priority/importance in graph layout.
+ */
+const DEFAULT_NODE_WEIGHTS = {
+  // Tier 1: Top-Level Orchestration & Scaling
+  HorizontalPodAutoscaler: 1000,
+
+  // Tier 2: Primary Workload Controllers
+  Deployment: 970,
+  StatefulSet: 970,
```

Comment:
stateful set and daemon set should have lower weight, deployment is more 'important'

Reply:
please double check the order of the resources here, so the order from 'main' branch is preserved

```tsx
const partitionLayers = [
  ['Deployment'],
  ['ReplicaSet', 'ServiceAccount', 'CronJob', 'DaemonSet', 'StatefulSet'],
  ['Job'],
  ['Pod', 'RoleBinding'],
  ['Service', 'NetworkPolicy', 'Role'],
  ['Endpoints'],
];
```

Reply:
fixed this